modules/library/metadata/src/main/java/org/geotools/util/factory/FactoryRegistry.java

```diff
@@ -453,25 +453,27 @@ private static void debug(
             final Hints.Key key,
             final String message,
             final Class<?> type) {
-        final StringBuilder buffer = new StringBuilder(status);
-        buffer.append(Utilities.spaces(Math.max(1, 7 - status.length())))
-                .append('(')
-                .append(Classes.getShortName(category));
-        if (key != null) {
-            buffer.append(", ").append(key);
-        }
-        buffer.append(')');
-        if (message != null) {
-            buffer.append(": ").append(message);
-        }
-        if (type != null) {
-            buffer.append(' ').append(Classes.getShortName(type)).append('.');
+        if (LOGGER.isLoggable(DEBUG_LEVEL)) {
```

Comment:
Nice refactor to avoid log data building if log level required is not reached.

Reply:
Eh yeah, was building a very large string for nothing, every single time.